### ISSUE TITLE
Content Creator Engine MVP

### ISSUE THREAD

**Issue by frogg** [OasysFoundation/Oasys-Organize#23]
Sketch out how an MVP of the content creation would work.

**Comment by dlrenz** [comment]
I think this is a great second step.
For the MVP we could try something simpler, for example an extended version of:
https://h5p.org/

**Comment by frogg** [comment]
Yep, I agree. Also in the math equation visualizer that @cocokiri is currently working on could be a good first MVP (see #5).
My sketch is the next step / long term-vision

**Comment by RobZuazua** [comment]
so what is the MVP for this and what is ETA?

**Comment by dlrenz** [comment]
Markus wanted to investigate h5p, and how to implement a math equation plugin

**Comment by dlrenz** [comment]
Here are a couple of interesting software projects that could be used as base:
https://quilljs.com/
https://prosemirror.net/
https://trix-editor.org/
https://alex-d.github.io/Trumbowyg/

**Comment by frogg** [comment]
In order to close this issue we need a clear document that outlines/combines everything that was posted in this issue. Deferring to next milestone to do so.

**Comment by RobZuazua** [comment]
@frogg @cocokiri ETA Tuesday for an MVP sounds great. How useful and helpful would this tool be for content creators? Can you paint a picture of how and when this would be useful for us?

**Comment by dlrenz** [comment]
I pushed a working minimal editor to our github toolbox. it works with quilljs and has basic text formatting and formula editing. 
Now we just need to add Markus's math plotting tool.

**Comment by frogg** [comment]
@dlrenz and @cocokiri will think about more extensions that make sense.

**Comment by RobZuazua** [comment]
@dlrenz can you link to the tool here so we can check it out?

**Comment by frogg** [comment]
@RobZuazua I can actually start hosting it on something like create.joinoasys.org once we have a MVP MVP!

Yesterday I managed to put the graph plot JS lib inside of QuillJS which seems to be promising for our MVP:

8c14f1f6eb8c1436e61337d2ef71725b3be27d9a

**Comment by dlrenz** [comment]
To make this really good, we should also have a look at what the competition implements, for example:
Smart sparrow
Acrobatiq

**Comment by dlrenz** [comment]
One important aspect would be to expose an API, e.g.
https://github.com/SmartSparrow/simcapi-js

**Comment by frogg** [comment]
We're still working on it.

**Comment by frogg** [comment]
To store the content, we can call `quill.getContents();`

Which gives us a json file describing the editor's content:
```json
{"insert": "Hello this is a test of the quill editor↵This is another "},
{"attributes": {"bold": true}, "insert": "test"},
{"insert": "↵↵↵↵↵"},
{"insert": {"graph": {"alt": null, "url": null}}}
```

Accordingly, we can set the content with `quill.setContents("JSON");` to edit an exiting document.

**Comment by frogg** [comment]
This needs to store the node UUID(s) as well.
